lambda/src/server/slack_db_linking.cljs

```diff
@@ -0,0 +1,151 @@
+(ns server.slack-db-linking
+  (:require [server.tokens :as tokens]
+            [server.firebase :as fire]
+            [kitchen-async.promise :as p]
+            [server.slack :as slack]
+            [cljs.pprint :as pp]
+            [applied-science.js-interop :as j]))
+
+;; LINKING
+;;
+;; Firebase schema -
+;; /slack-channel/{id}   => {project-id, workspace-id}
+;; /slack-user/{id}      => {account-id, workspace-id}
+;; /slack-workspace/{id} => {board-id}
+
+(defn slack-user-by-email
+  "Returns slack user id for the given email address, if found"
+  [email]
+  (p/let [res (slack/get+ "users.lookupByEmail" {:query {:email email}})]
+    ;; UNCLEAR: how does Slack know what workspace we are interested in?
```

Comment:
I believe this is handled by our API token, which means 1. we'll need to manually set up each Slack workspace with a bot, which I think we knew, but 2. we'll need to manage a map of these in our config.

Reply:
Sorry — does "workspace" in this context mean Slack workspace or Firebase/SparkBoard workspace?

Reply:
workspace is always slack workspace. maybe we should pass maps around and use namespaced keywords because otherwise this will get confusing quickly?